File: Lessons/Lesson 31. Project 21. Clean Swift. Part 1/Outline/Snow.dog/recruitment-test-ios/iOSRecruitmentTest/Item.swift
//
//  Item.swift
//  iOSRecruitmentTest
//
//  Created by msm72 on 7/23/16.
//  Copyright © 2016 Snowdog. All rights reserved.
//

import Foundation
import CoreData
import Gloss

class Item: NSManagedObject, Decodable {
    // MARK: - Initialization
    override init(entity: NSEntityDescription, insertIntoManagedObjectContext context: NSManagedObjectContext?) {
        super.init(entity: entity, insertIntoManagedObjectContext: context)
    }
    
    required init?(json: JSON) {
        super.init(entity: CoreDataManager.instance.entityForName("Item"), insertIntoManagedObjectContext: CoreDataManager.instance.managedObjectContext)
        
        parsing(json)
    }
    
    
    // MARK: - Custom Functions
    func parsing(json: JSON) {
        self.id = ("id" <~~ json)!
        self.name = ("name" <~~ json)!
        self.comment = ("description" <~~ json)!
        self.imageURL = ("icon" <~~ json)!
    }
}

File: Lessons/Lesson 31. Project 21. Clean Swift. Part 1/Outline/Snow.dog/npm/server-for-mobile-recruitment/models/item.js
var item = function (i, timestamp) {
  this.id = i;
  this.name = "Item (x)".replace(/x/g, i);
  this.description = "Description of item (x)".replace(/x/g, i);
  this.icon = "/x/icon.png".replace(/x/g, i);
  this.timestamp = timestamp;
  this.url = "/x".replace(/x/g, i);
};

module.exports = item;
